### PR TITLE
Add Powerpoint Slide Loader PptxSlideReader to LLama-hub

### PR DESCRIPTION
**Pull request by tewnut**
# Description
This is a new loader to read a PowerPoint presentation file and create a list of slide Documents.

## Type of Change
Please delete options that are not relevant.
- [x] New Loader/Tool. 

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] Added new notebook (that tests end-to-end). [Link to the notebook](https://colab.research.google.com/drive/14SQJ-Sccgpt1U_q5_dmj6uTrKXZbZTAN?usp=sharing)

# Suggested Checklist:
- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by nerdai** [comment]
Thanks @tewnut for the contribution! It looks though that we already have `PptxReader`.  Is there a reason in your mind for us to include this new one?

**Comment by tewnut** [comment]
> Thanks @tewnut for the contribution! It looks though that we already have `PptxReader`. Is there a reason in your mind for us to include this new one?

It's exactly why I have included a FAQ:
---
**What is the difference with file/pptx loader?**
The file/pptx loader creates one Document that joins all the slides. In contrast, this file/pptx-slide loader creates a list of Documents corresponding to slides in the presentation.
---
If I make a change to the original file/pptx loader, I bet lots of applications depending on it will break due to output differences.

**Comment by nerdai** [comment]
> > Thanks @tewnut for the contribution! It looks though that we already have `PptxReader`. Is there a reason in your mind for us to include this new one?
> 
> It's exactly why I have included a FAQ: **What is the difference with file/pptx loader?** The file/pptx loader creates one Document that joins all the slides. In contrast, this file/pptx-slide loader creates a list of Documents corresponding to slides in the presentation.
> 
> If I make a change to the original file/pptx loader, I bet lots of applications depending on it will break due to output differences.

Awesome, thanks @tewnut for being so thoughtful on this -- much appreciated!

**Review by nerdai — APPROVED** [review]
Just needs to pass our checks, then it can be merged!

File: llama_hub/file/pptx-slide/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+documents = loader.load_data(
+  file=Path('./deck.pptx'),
+  extra_info={"source" : "my-deck.pptx"},
+  join_char='\n',
+  include_shapes=TRUE
+)
+# Alternatively: documents = loader.load_data(file='./deck.pptx')
+```
+
+This loader is designed to be used as a way to load data into [LlamaIndex](https://github.com/run-llama/llama_index/tree/main/llama_index) and/or subsequently used as a Tool in a [LangChain](https://github.com/hwchase17/langchain) Agent. See [here](https://github.com/emptycrown/llama-hub/tree/main) for examples.
```
Comment (anoopshrma) [inline llama_hub/file/pptx-slide/README.md:30]: Could you update the llama-hub repo link to the updated link  here!

Replace it with: https://github.com/run-llama/llama-hub/tree/main/llama_hub

Thanks!!

**Review by anoopshrma — APPROVED** [review]
Just added one comment, Rest everything looks super fine @tewnut 

Thank you for the contribution!!

File: llama_hub/file/pptx-slide/README.md (hunk 1 above)
Comment (tewnut) [reply llama_hub/file/pptx-slide/README.md]: README is updated with official links

**Comment by tewnut** [comment]
> Just needs to pass our checks, then it can be merged!

ruff/black checks passed.

**Pull request merged**